pkg/microservice/warpdrive/core/service/taskplugin/deploy.go

```diff
@@ -544,6 +544,47 @@ func (p *DeployTaskPlugin) Run(ctx context.Context, pipelineTask *task.Task, _ *
 			Wait:        true,
 		}
 
+		failedStatusProcess := func() error {
+			hrs, errHistory := helmClient.ListReleaseHistory(chartSpec.ReleaseName, 5)
+			if errHistory != nil {
+				return fmt.Errorf("failed to ListReleaseHistory : %s,error:%s", chartSpec.ReleaseName, errHistory)
+			}
+			if len(hrs) > 0 {
+				releaseutil.Reverse(hrs, releaseutil.SortByRevision)
+				rel := hrs[0]
+				if rel.Info.Status == helmrelease.StatusDeployed {
+					return nil
+				}
```

Comment:
this if statement can be removed, since rel.Info.Status == helmrelease.StatusFailed will be used later